jib-core/src/main/java/com/google/cloud/tools/jib/blob/Blobs.java

```diff
@@ -77,5 +80,31 @@ public static String writeToString(Blob blob) throws IOException {
     return byteArrayOutputStream.toByteArray();
   }
 
+  /**
+   * Gets a {@link Blob} that is {@code blob} compressed.
+   *
+   * @param blob the {@link Blob} to compress
+   * @return the compressed {@link Blob}
+   */
+  public static Blob compress(Blob blob) {
+    return Blobs.from(
+        outputStream -> {
+          try (GZIPOutputStream compressorStream = new GZIPOutputStream(outputStream)) {
+            blob.writeTo(compressorStream);
+          }
+        });
+  }
+
+  /**
+   * Gets a {@link Blob} that is {@code blob} decompressed.
+   *
+   * @param blob the {@link Blob} to decompress
+   * @return the decompressed {@link Blob}
+   * @throws IOException if an I/O exception occurs
+   */
+  public static Blob decompress(Blob blob) throws IOException {
+    return Blobs.from(new GZIPInputStream(new ByteArrayInputStream(Blobs.writeToByteArray(blob))));
```

Comment:
This was reasonable in test code, but I expect there must be a more efficient way for production. It is very efficient to first write all the blob content to an array in memory and then have `GZIPInputStream` to read the content out of the array. We should leverage stream and definitely avoid writing holding the content in memory.

Reply:
This isn't being used anywhere so I might just move this back to the test class.